docs/TDD.md

```diff
@@ -0,0 +1,220 @@
+If the tests pass, then either implementing the behavior is done or perhaps more necessary tests remain to be created.
+If the tests don't pass, then the code is debugged and the tests are run again.
+The cycle of testing and coding is repeated until all of the necessary tests pass, at which point the implementation for that aspect of the program's behavior is done... for now.
+
+## Refactoring
+
+[Refactoring][refactoring] is the rewriting of code to improve its design.
+It is not simply rewriting code to fix bugs.
+It is sometimes called "refactoring" to modify code to get it to pass the tests.
+Although modifying the code _may_ include improving the design as a way to pass the tests, simply debugging is not necessarily improving the _design_ of the code, and so is not necessarily _refactoring_.
+
+Following is an example of debugging without refactoring:
+
+```python
+
+# A function intended to return x added to y.
+# x and y are bad parameter names, but we ignore that for now.
+def add(x, y):
+    # used multiply operator by mistake. It fails the tests.
+    return x * y
+
+# Function corrected. It passes the tests. It has been debugged, but not refactored.
+def add(x, y):
+    return x + y
+    
+```
+
+
+Following is an example of refactoring, and then debugging:
+
+```python
+
+# Function name and parameter names are modified to something more meaningful. This is refactoring.
+def lot_inventory(old_cars, new_cars):
+    # Introduced multiply operator by mistake. It fails the tests. This is why we test.
+    return old_cars * new_cars
+
+# Function corrected. It passes the tests. This is debugging.
+def lot_inventory(old_cars, new_cars):
+    return old_cars + new_cars
+
+```
+
+## TDD and Python on Exercism
+
+Exercism's Python track utilizes TDD methodology in its exercises.
+Unit tests are already written.
+The tests may be viewed by the student to gather a more detailed understanding of what a passing solution should provide.
+A solution stub may be provided to the student.
+
+## Troubleshooting a Failed Test on Exercism in the Web Editor
+
+When one or more tests fail for a Python solution, the corresponding task(s) will not have a green background.
+The first failing task area will be expanded and its header will look something like
+
+```
+Task 1 Extract coordinates -
+```
+
+Clicking on the minus sign will collapse the task, so we can look at other tasks, but for now we will stick with this task.
+
+Underneath will be an expanded Test area that will look something like
+
+```
+       Test 1                               ⌄
+FAILED TisburyTreasure > get coordinate
+```
+
+where `Tisbury Treasure` indicates the exercise, and `get_coordinate` indicates the failing function or method.
+
+`Test 1` is usually going to be a kind of template with a code section for setting up the tests.
+It does not have information on the specific test(s) that failed. 
+It will say toward the bottom that 
```

Comment:
```suggestion
It does not have information on the specific test(s) that failed.
It will say toward the bottom that
```

Trailing white space, but the second line looks as though it is incomplete.